src/network.rs

```diff
@@ -857,6 +865,27 @@ impl<'a> Network<'a> {
     };
   }
 
+  async fn user_artist_check_follow(&mut self) {
+    let mut app = self.app.lock().await;
```

Comment:
Ah just noticed this!

One problem with this new threaded async architecture is that if we aquire the `lock` _before_ the async action, we can end up blocking the UI rendering loop (since the start of the loop also aquires a `lock`).

This means we must be careful to only the the `lock` _after_ the async action is finished (i.e. when we need to update `app`).

To fix, we must dispatch the data needed for the network request like this

```rust
// From some handler somewhere
 if let Some(artists) = &app.search_results.artists {
      let artist_ids: Vec<String> = artists
        .artists
        .items
        .iter()
        .map(|item| item.id.to_owned())
        .collect();

      app.dispatch(IoEvent::UserArtistFollowCheck(artist_ids));
}

// in network.rs
async fn user_artist_check_follow(&mut self, album_ids: Vec<String>)
```

You can see this pattern in the other handlers.

This is an annoying gotcha that I'd like to solve somehow

Reply:
Fixed. However, this change cause one problem.
Just after I searched with one keyword, no albums or artists are marked up even if they are saved or followed and I searched with the same keyword, they are marked up successfully.
Ithink that artsis and album ids were `None` at the first searching because `get_search_result` did not finished at that time.

Reply:
In previous commit, id vector is generated after `get_search_result` finished and such problem did not occur. I'm not sure which is better so please check the problem and make decision.